lib/cli.js

```diff
@@ -178,7 +189,13 @@ async function translateOptions({
         ignore,
         overrideConfig,
         overrideConfigFile,
-        reportUnusedDisableDirectives: reportUnusedDisableDirectives ? "error" : void 0
+        reportUnusedDisableDirectives: reportUnusedDisableDirectives ? "error" : void 0,
+
+        /*
+         * For performance reasons rules not marked as 'error' are filtered out in quiet mode. As maxWarnings
+         * requires rules set to 'warn' to be run, we only filter out 'warn' rules if maxWarnings is 0.
+         */
+        ruleFilter: quiet && maxWarnings === -1 ? quietRuleFilter : () => true
```

Comment:
Since only the `FlatESLint` class accepts this option, we should set it below, inside `if (configType === "flat")`.